### ISSUE TITLE
Decouple Java class construction from frameworks

### ISSUE THREAD

**Issue by OneCricketeer** [operator-framework/java-operator-plugins#71]
#### Describe the problem you need a feature to resolve.

It seems the only way to use this is to rely on some additional Quarkus boilerplate (at least a properties file)

#### Describe the solution you'd like.

I'd like to just construct some POJOs from operator-sdk without needing to additionally cleanup extraneous resources that aren't required if I am not interested in using Quarkus.

**Comment by OneCricketeer** [comment]
The "quarkus app-type" generator could construct a multi-module project that depends on the proposed "model" classes

**Comment by metacosm** [comment]
Out of curiosity, are you planning on using any other framework or do you plan on using plain Java?

**Comment by OneCricketeer** [comment]
Currently using Spring, but I don't expect to be needing Autowire annotations or anything like that.

**Comment by OneCricketeer** [comment]
So, I already have an operator generated from Golang `operator-sdk`

I found this repo while searching for this solution, thinking it was the same idea rather than just having this repo be a sort of project initializer - https://github.com/fabric8io/kubernetes-client/pull/3693

**Comment by csviri** [comment]
Having some target environment / framework flag probably might help.  Also the generator could be integrated. 

I idea was the same, just the implemented features are not there yet.

**Comment by metacosm** [comment]
OK, so the idea is to port your operator to Java? It would be interesting to hear about your use case if you could talk about it, that is :)

**Comment by OneCricketeer** [comment]
@metacosm No, at least, not that I've heard. We have it written and deployed from the Go model, but have a Java API layer that constructs CRD payload representations, passes them around on some message queues, writes them to databases, etc. We currently are copying attributes between the two models manually, thus why I was searching for code generation utilities.
